predicates/forwarded/forwarded_test.go

```diff
@@ -0,0 +1,308 @@
+package forwarded
+
+import (
+	"net/http"
+	"net/url"
+	"testing"
+)
+
+type request struct {
+	url     string
+	headers http.Header
+}
+
+func TestForwardedHost(t *testing.T) {
+	testCases := []struct {
+		msg     string
+		host    string
+		r       request
+		matches bool
+		isError bool
+	}{{
+		msg:     "Empty host should fail",
+		host:    "",
+		r:       request{},
+		matches: false,
+		isError: true,
+	}, {
+		msg:     "Empty Forwarded Header should not match",
+		host:    "example.com",
+		r:       request{},
+		matches: false,
+		isError: false,
+	}, {
+		msg:  "Same forwarded Header should match",
+		host: "example.com",
+		r: request{
+			url: "https://myproxy.com/index.html",
+			headers: http.Header{
+				"Forwarded": []string{"for=192.0.2.60;proto=http;by=203.0.113.43;host=example.com"},
+			},
+		},
+		matches: true,
+		isError: false,
+	}, {
+		msg:  "Same forwarded Header should match",
+		host: "example.com",
+		r: request{
+			url: "https://myproxy.com/index.html",
+			headers: http.Header{
+				"Forwarded": []string{"for=192.0.2.60;proto=http;host=example.com;by=203.0.113.43"},
+			},
+		},
+		matches: true,
+		isError: false,
+	}, {
+		msg:  "Same forwarded Header should match",
```

Comment:
I think this one is redundant as there should be three "Same forwarded Header should match"  where host is at the start, middle and end of the header.